tests/test_steps.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env python
+from datetime import datetime
+from pathlib import Path
+import shutil
+import subprocess
+import tarfile
+from tempfile import mkdtemp
+
+
+def checkpaths(paths):
+    result = [Path(pth).is_file() for pth in paths]
+    print("\n")
+    # report which files are missing (in red) and which are present (in green)
+    for pth, res in zip(paths, result):
+        if res:
+            print(f"\033[92m{pth} is present\033[0m")
+        else:
+            print(f"\033[91m{pth} is missing\033[0m")
+    return all(result)
```

Comment:
Since you're looping through these anyway, it might be simpler to do this check within the for-loop itself
```suggestion
    print("\n")
    # report which files are missing (in red) and which are present (in green)
    for pth in paths:
        if Path(pth).is_file():
            print(f"\033[92m{pth} is present\033[0m")
        else:
            print(f"\033[91m{pth} is missing\033[0m")
    return all(result)
```

Reply:
Thanks for this suggestion @hollandjg. Note that I am using the `result` list as the return object in combination with the `all` function. The suggested approach would require creating an empty list for `result` and populate it within the for loop  that adds the convenient side effect of displaying what files are present/absent with color. I am trying to avoid that pattern with the list comprehension, which is the pythonic way of performing such tasks. What do you think?

Reply:
Oh I see. Makes sense!